### PR TITLE
feat(react-tags): make basic Tag a button instead of div

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
### Description

Before this PR, basic Tag has `dismissButton` slot to dismiss itself. This PR make the Tag itself a button that does dismiss, and changes `dismissButton` slot to `dismissIcon`. And this PR keeps TagButton unchanged - it's still two buttons.

#### before
```tsx
root: NonNullable<Slot<'div'>>;
dismissButton?: Slot<'button'>;
```
image

#### after
```tsx
root: NonNullable<Slot<'button'>>;
dismissIcon?: Slot<'span'>;
```
image

- Why this change:
In fluent v0 Pill, the focus ring is around the entire Pill. We would want the same for v9 basic Tag. Also this is useful for user with magnifier that tracks focus.

- What does this mean for #27794:
I will keep two controls Tag/TagButton instead of one Tag, and close PR 27794. Because basic Tag and TagButton have different slots. The differences are:
    1. Basic Tag has dismiss icon (`Slot<span>`), while TagButton has dismiss button (`Slot<button>`)
    2. TagButton has `content` slot that wraps around media/primary and secondary text and act as first tab stop. Basic Tag does not need `content` slot.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/322ee9ad8bbd87f7721068f664bfb530baf56623'>322ee9ad8bbd87f7721068f664bfb530baf56623</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts [hunk 1]
```diff
@@ -15,39 +15,68 @@ export const tagButtonClassNames: SlotClassNames<TagButtonSlots> = {
...
+    "media secondary"
+    "media .        "
+    `,
+    paddingRight: tokens.spacingHorizontalS,
+
     ...createCustomFocusIndicatorStyle(
       {
         ...shorthands.borderRadius(tokens.borderRadiusMedium),
         ...shorthands.outline(tokens.strokeWidthThick, 'solid', tokens.colorStrokeFocus2),
+        zIndex: 1,
```
Comment (ling1726) [inline packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts:49]: is the z-index necessary?

File: packages/react-components/react-tags/src/utils/useTagContextValues.ts [hunk 2]
```diff
@@ -1,7 +1,10 @@
 import * as React from 'react';
 import { TagContextValues, TagState } from '../components/Tag/index';
+import { TagButtonContextValues, TagButtonState } from '../TagButton';
 
-export function useTagContextValues_unstable(state: TagState): TagContextValues {
+export function useTagContextValues_unstable(
+  state: TagState | TagButtonState,
```
Comment (ling1726) [inline packages/react-components/react-tags/src/utils/useTagContextValues.ts:6]:
```suggestion
  state: Pick<TagState, 'avatarSize' | 'avatarShape'>
```
nit: that should still work without needed to union both state types

**Review by bsunderhus — APPROVED** [review]
LGTM

**Pull request merged**